crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -91,36 +91,40 @@ pub struct NotificationSettings {
     rules: Arc<RwLock<Rules>>,
     /// Drop guard of event handler for push rules event.
     _push_rules_event_handler_guard: Arc<EventHandlerDropGuard>,
+    /// Notified every time the push rules change, either due to sync or local
+    /// changes.
     changes_sender: broadcast::Sender<()>,
 }
 
 impl NotificationSettings {
-    /// Build a new `NotificationSettings``
+    /// Build a new [`NotificationSettings`].
     ///
     /// # Arguments
     ///
-    /// * `client` - A `Client` used to perform API calls
-    /// * `ruleset` - A `Ruleset` containing account's owner push rules
+    /// * `client` - A [`Client`] used to perform API calls.
+    /// * `ruleset` - A [`Ruleset`] containing account's owner push rules.
     pub(crate) fn new(client: Client, ruleset: Ruleset) -> Self {
         let changes_sender = broadcast::Sender::new(100);
         let rules = Arc::new(RwLock::new(Rules::new(ruleset)));
 
-        // Listen for PushRulesEvent
+        // Listen for PushRulesEvent.
         let push_rules_event_handler_handle = client.add_event_handler({
             let changes_sender = changes_sender.clone();
-            let rules = Arc::clone(&rules);
+            let rules = rules.clone();
```

Comment:
I was under the impression that `Arc::clone` is more idiomatic. I do find that style here: https://doc.rust-lang.org/stable/book/ch16-03-shared-state.html and here: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.clone . Does the house style do it differently?